### PR TITLE
Quickicon plugin loader

### PR DESCRIPTION
**Pull request by bembelimen**
Improve the dashboard modules.

### CODE REVIEW

**Comment by brianteeman** [comment]
@bembelimen I was about to report that the quickicons showed the spinner icon when the count value was 0 but then I saw this PR. After using the js changes here I can confirm that Articles and Checkout now correctly show a 0 but cache is showing false

image

**Comment by bembelimen** [comment]
Additional it should also fix the plugin loading issue...I still have to add one or two improvements but in general that could be the way for the icons.

**Comment by brianteeman** [comment]
I only checked 
Fix loader with "0" as value

**Review by lavipr — APPROVED** [review]

File: administrator/modules/mod_quickicon/Helper/QuickIconHelper.php [hunk 1]
```diff
@@ -53,194 +55,235 @@ public static function &getButtons(Registry $params, CMSApplication $application
...
-				PluginHelper::importPlugin('quickicon', 'joomlaupdate');
-			}
+			self::$buttons[$key] = [];
 
-			if ($params->get('show_eupdate'))
-			{
-				PluginHelper::importPlugin('quickicon', 'extensionupdate');
-			}
+			// @TODO: Perhaps we should use something with EventAware
+			$container = (new Container)
```
Comment (mbabker) [inline administrator/modules/mod_quickicon/Helper/QuickIconHelper.php:67]: What are you doing here?  You're creating a new DI container and creating a new event dispatcher.  For what purpose?  There is exactly one event dispatcher in a Joomla application.  I assume this is for your "scope" of distinguishing different contexts or "core" versus "third party", these are the types of checks that belong in the plugin, not bypassing the global event dispatcher.

The entire redesign of this module is beyond wrong.  There should not be a distinction between core extensions and third party extensions.  What makes something core versus third party?  If someone adds a quickicon for weblinks, is that core or third party?  Why should a module control what plugins are or are not loaded, this is the type of behavior that needs to be erased from Joomla core code (expanded on below).

There should never be any code added in core for specific plugins.  This means that the capability is not something that should be a plugin in the first place.  This has been a problem in core for a long time and requires effort to fix, there should not be new code introduced using this anti-pattern.  Therefore, the options in this module to show or hide the result from a plugin should be removed, or the plugins should be removed and their capabilities inlined into this module.

And as @brianteeman has said about a thousand times, this is the type of change that does not belong in this repository but should have been made in the main repository.  There has been a massive lack of code review in the quickicon module changes and the result is more people are going to have to invest more time into fixing things, assuming they can discover the problems.

Comment (brianteeman) [reply administrator/modules/mod_quickicon/Helper/QuickIconHelper.php:67]: I am sure I only said it a hundred times not a thousand ;)

**Pull request merged**